### ISSUE TITLE
Text & elements not allowed inside SVG's `<metadata>`

### ISSUE THREAD

**Issue by tigt** [validator/validator#156]
**Test case:**
```html
<!DOCTYPE html>
<html>
<head>
  <title>Content inside SVG's &lt;metadata&gt;</title>
</head>
<body>
  <svg>
    <metadata>I <em>don't</em> validate!</metadata>
  </svg>
</body>
</html>
```

**What happens:**
>**Error:** Text not allowed in element `metadata` in this context.
>**Error:** HTML start tag `em` in a foreign namespace context.

**What should happen:**
`<metadata>` was designed as a dumping ground for any sort of (ostensibly XML) data that other programs wanted to embed in SVG, so nothing other than `</metadata>` should be disallowed.

>The contents of the 'metadata' should be elements from other XML namespaces, with these elements from these namespaces expressed in a manner conforming with either the Namespaces in XML 1.0 or Namespaces in XML 1.1 Recommendations [XML-NS10][XML-NS].

With HTML5's foreign content parsing rules, I think this translates to "don't leave elements unclosed."

Possibly useful links:
* [HTML5 8.2.5.4 - The rules for parsing tokens in foreign content](http://www.w3.org/html/wg/drafts/html/master/syntax.html#parsing-main-inforeign)
* [SVG Tiny 1.2 - The metadata element](http://www.w3.org/TR/SVGTiny12/metadata.html#MetadataElement)

**Comment by sideshowbarker** [comment]
So yeah after doing some testing I realize that what you’re seeing here is strictly a consequence of the SVG parsing behavior defined in the HTML spec. The HTML parsing algorithm actually does handle text within the SVG `metadata` element just as you’d want; `<svg><metadata>I validate!</metadata></svg>` parses and validates just fine.

What the HTML parsing algorithm doesn’t handle in the way you want is that if put an HTML element such as `em` as a child of an SVG `metadata` element in your source, you don’t get the DOM you want—in that the the `em` does not become a child of the `metadata` element in the DOM; instead, it gets hoisted all the way out to become a following sibling of the `svg` ancestor of that `metadata` element.

And the http://www.w3.org/html/wg/drafts/html/master/syntax.html#parsing-main-inforeign section of the spec you cite spec defines that `metadata`/`em` case as a parse error.

Try your HTML source snippet in http://software.hixie.ch/utilities/js/live-dom-viewer/ or even just in the DOM view of any browser built-in devtools/inspector and you’ll see what I mean.

So if you want to be able to use HTML elements inside the SVG `metadata` element, what you’d need to do is to get the HTML spec changed so that the HTML parsing algorithm treats `metadata` differently, and all browser-engine implementations would need to be changed to match the HTML spec change.

But It is extremely unlikely that browser-engine implementations would agree at this point to change their parsing behavior around this, so IMHO it wouldn’t be worth pursing. But if you did want to try it, the thing to do would be to file a new issue at https://github.com/whatwg/html

Anyway, as far as this issue here goes, I would suggest we agree to close it, because the checker is behaving as expected here, in conformance with the requirements in the HTML spec.

**Comment by tigt** [comment]
Works for me. `<script type="what/ever+xml">` probably works better for arbitrary data embedding anyway.

**Comment by sideshowbarker** [comment]
> Works for me. `<script type="what/ever+xml">` probably works better for arbitrary data embedding anyway.

Good point, yeah

**Comment by ahtashamilyas** [comment]
Not work on my side. Got error "HTML start tag 'span' in a foreign namespace context". inside SVG.
